context/app/static/js/pages/Template/Template.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useCallback } from 'react';
+import Stack from '@mui/material/Stack';
+import AccordionDetails from '@mui/material/AccordionDetails';
+import Button from '@mui/material/Button';
+import Chip from '@mui/material/Chip';
+import Typography from '@mui/material/Typography';
+import KeyboardArrowDownRoundedIcon from '@mui/icons-material/KeyboardArrowDownRounded';
+
+import { useWorkspaceTemplates } from 'js/components/workspaces/NewWorkspaceDialog/hooks';
+import SummaryPaper from 'js/shared-styles/sections/SectionPaper';
+import SummaryData from 'js/components/detailPage/summary/SummaryData';
+import LabelledSectionText from 'js/shared-styles/sections/LabelledSectionText';
+import LogInPanel from 'js/shared-styles/panels/LogInPanel';
+import { InternalLink } from 'js/shared-styles/Links';
+import IconPanel from 'js/shared-styles/panels/IconPanel';
+import WorkspaceDatasetsTable from 'js/components/workspaces/WorkspaceDatasetsTable';
+import { StyledAccordionSummary } from 'js/pages/Template/style';
+import { StyledAccordion } from 'js/components/workspaces/AdvancedConfigOptions/style';
+import { NewWorkspaceDialogFromExample } from 'js/components/workspaces/NewWorkspaceDialog';
+import { useCreateWorkspaceForm } from 'js/components/workspaces/NewWorkspaceDialog/useCreateWorkspaceForm';
+import { useDatasetTypeMap } from 'js/components/home/HuBMAPDatasetsChart/hooks';
+import { TemplateExample } from 'js/components/workspaces/types';
+
+interface ExampleAccordionProps {
+  example: TemplateExample;
+  templateKey: string;
+  datasetTypeMap: Record<string, string[]>;
+  defaultExpanded?: boolean;
+}
+
+function ExampleAccordion({ example, templateKey, datasetTypeMap, defaultExpanded }: ExampleAccordionProps) {
+  const { title, description, assay_display_name, datasets, job_type, resource_options } = example;
+
+  const { setDialogIsOpen, ...rest } = useCreateWorkspaceForm({
+    defaultName: title,
+    defaultTemplate: templateKey,
+    defaultJobType: job_type,
+    defaultResourceOptions: resource_options,
+    initialSelectedDatasets: datasets,
+  });
+
+  const getRawDatasetType = useCallback(
+    (name: string) => Object.keys(datasetTypeMap).find((key) => datasetTypeMap[key].includes(name)) ?? name,
+    [datasetTypeMap],
+  );
```

Comment:
Might make more sense to create a map of the `assay_display_name` : `raw_dataset_type`. @NickAkhmetov what do you think?

Reply:
That approach makes sense to me, i.e. creating that mapping in a `useMemo` and using that for lookups should decrease the complexity somewhat since `Object.keys` and `find` are both `O(n)` operations.